chia-traits/src/streamable.rs

```diff
@@ -115,9 +115,8 @@ impl<T: Streamable> Streamable for Vec<T> {
     fn parse(input: &mut Cursor<&[u8]>) -> Result<Self> {
         let len = u32::parse(input)?;
 
-        // TODO: pre-allocate capacity, but we'd need safe-guards for overflow
-        // attacks
-        let mut ret = Vec::<T>::new();
+        let limit = 2 * 1024 * 1024 / std::mem::size_of::<T>();
```

Comment:
Not sure if it matters in this context, but if the type is zero-sized, this will panic. In this edge case, the capacity should default to `usize::MAX`, since no allocation is required.

Reply:
right. I was considering this case. I assume it would fail to compile, since this is a constant expression. (which I think is fine)

Reply:
I think it would need to be marked as `const` rather than `let` to be evaluated at compile-time

Reply:
it appears `const` can't depend on template parmeters in rust. I get:
```
error[E0401]: can't use generic parameters from outer item
   --> chia-traits/src/streamable.rs:118:68
    |
95  | impl<T: Streamable> Streamable for Vec<T> {
    |      - type parameter from outer item
...
118 |         const LIMIT: usize = 2 * 1024 * 1024 / std::mem::size_of::<T>();
    |                                                                    ^ use of generic parameter from outer item
```